pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -1192,6 +1327,9 @@ public static PinotDataType getSingleValueType(Class<?> cls) {
     if (cls == String.class) {
       return STRING;
     }
+    if (cls == BigDecimal.class) {
```

Comment:
(minor) Move it after `Timestamp` as it should be rarely used as the input object